applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -77,20 +166,22 @@ static void handle_dfu_data(const struct config_event *event)
 		return;
 	}
 
-	LOG_INF("DFU data received cur_offset:%" PRIu32, cur_offset);
+	LOG_DBG("DFU data received cur_offset:%" PRIu32, cur_offset);
 
 	if (size == 0) {
 		LOG_WRN("Invalid DFU data header");
 		goto dfu_finish;
 	}
 
 	if (is_new_page_started(cur_offset, size)) {
-		err = flash_area_erase(flash_area,
-		       FLASH_PAGE_ID(cur_offset + size - 1) << FLASH_PAGE_SIZE_LOG2,
-		       FLASH_PAGE_SIZE);
-		if (err) {
-			LOG_ERR("Cannot erase page (%d)", err);
-			goto dfu_finish;
+		u32_t off = FLASH_PAGE_ID(cur_offset + size - 1) << FLASH_PAGE_SIZE_LOG2;
+
+		if (!is_page_clear(flash_area, off, FLASH_PAGE_SIZE)) {
+			err = flash_area_erase(flash_area, off, FLASH_PAGE_SIZE);
```

Comment:
can this happen in parallel with background erase?

Reply:
This should not happen in parallel with background erase (during background erase DFU is inactive).